yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -228,6 +228,7 @@ private[spark] class ApplicationMaster(
   }
 
   private def sparkContextStopped(sc: SparkContext) = {
+    sc.requestTotalExecutors(0, 0, Map.empty)
```

Comment:
i think   
 if (allocator != null) {
      allocator.requestTotalExecutorsWithPreferredLocalities(0, 0, Map.empty)
    }
is better that before.